Classes/Domain/TemplateNodeCreationHandler.php

```diff
@@ -56,7 +63,9 @@ public function handle(NodeInterface $node, array $data): void
             $template = $this->templateConfigurationProcessor->processTemplateConfiguration($templateConfiguration, $evaluationContext, $caughtExceptions);
             $this->exceptionHandler->handleAfterTemplateConfigurationProcessing($caughtExceptions, $node);
 
-            $this->nodeCreationService->apply($template, $node, $caughtExceptions);
+            $nodeMutators = $this->nodeCreationService->createMutatorsForRootTemplate($template, $node->getNodeType(), $this->nodeTypeManager, $node->getContext(), $caughtExceptions);
+            $nodeMutators->executeWithStartingNode($node);
+
             $this->exceptionHandler->handleAfterNodeCreation($caughtExceptions, $node);
```

Comment:
hm, I find it interesting that we use exceptions for control flow - and no "Result" objects or anything like that.

And this custom exception handler should be named differently IMHO, as I would expect a standard exception handler here.

Reply:
Will be fixed with https://github.com/Flowpack/Flowpack.NodeTemplates/pull/74